src/app/code/community/Shopgate/Cloudapi/Helper/Request.php

```diff
@@ -60,14 +70,59 @@ protected function parameterDetected()
      */
     protected function setCookie()
     {
-        Mage::getSingleton('core/cookie')->set(self::COOKIE_NAME, self::COOKIE_VALUE, 0);
+        $data = new Varien_Object();
+        $data->setData(self::KEY_SGCLOUD_INAPP, self::COOKIE_VALUE);
+        $data->setData(self::KEY_SGCLOUD_IS_CHECKOUT, $this->getParam(self::KEY_SGCLOUD_IS_CHECKOUT));
+        $data->setData(self::KEY_SGCLOUD_CALLBACK_DATA, $this->getParam(self::KEY_SGCLOUD_CALLBACK_DATA));
+
+        Mage::getSingleton('core/cookie')->set(self::COOKIE_NAME, json_encode($data->getData()), 0);
     }
 
     /**
+     * @param string $key
+     * @param mixed  $value
      * @return bool
      */
-    protected function cookieIsSet()
+    public function cookieIsSet($key, $value = false)
```

Comment:
The ```$value``` does not make sense based on the name of the class. That is, this function is doing more than it should be. If the logic is used that way, then there needs to be some refactoring.